src/module/utils/sheetUtils.ts

```diff
@@ -189,20 +188,19 @@ export function getKeyByValue(object:{ [x:string]:unknown; }, value:unknown):str
 
 export function getDataFromDropEvent(event:DragEvent):Record<string, any> {
   try {
-    return JSON.parse(event.dataTransfer.getData('text/plain'));
+    return JSON.parse(<string>event.dataTransfer?.getData('text/plain'));
   } catch (err) {
     throw new Error(game.i18n.localize("TWODSIX.Errors.DropFailedWith").replace("_ERROR_MSG_", err));
   }
 }
 
-export async function getItemDataFromDropData(data:Record<string, any>):Promise<TwodsixItemData> {
+export async function getItemDataFromDropData(data:Record<string, any>) {
   if (data.pack) {
     // compendium
     const pack = game.packs.find((p) => p.collection === data.pack);
-    if (pack.metadata.type !== 'Item') {
+    if (!pack || pack.metadata.entity !== 'Item') {
```

Comment:
I thought that packs have now migrated to "type" and not "entity" in v9

Reply:
Oh, right, I trusted the foundryvtt-types a bit too much there, I guess they haven't gotten around to updating that bit yet.